### ISSUE TITLE
Appearance changes ignored by color prop of iconBackground options

### ISSUE THREAD

**Issue by mattmcdonald-uk** [wix/react-native-navigation#7499]
## 🐛 Bug Report

The `color` property in `iconBackground` options doesn't respect appearance changes, either when using `PlatformColor` or when using an object with `light` and `dark` properties for the `color` value.

## To Reproduce

Create a button and use a `PlatformColor` value for the `iconBackground` colour.

```ts
Navigation.mergeOptions('componentId', {
    topBar: {
        leftButtons: [
            {
                color: PlatformColor('SurfaceForeground'),
                icon: {
                    uri: 'Close',
                },
                iconBackground: {
                    color: PlatformColor('Surface'),
                    cornerRadius: 40,
                    height: 40,
                    width: 40,
                },
            },
        ],
    },
})
```

Change the appearance (e.g. from dark to light) and observe that the icon colour updates, but the icon background colour does not.

As a workaround you can wrap this in `useEffect` with `colorScheme` from `useColorScheme` as a dependency - this isn't ideal though as there's a flicker with this colour change being out of sync with others.

This is with React Native 0.67.4 and iOS 15.4 with RNN 7.26.0

**Comment by mattmcdonald-uk** [comment]
This is still an active issue

**Comment by mattmcdonald-uk** [comment]
@yogevbd Could you please re-open this issue?

**Comment by mattmcdonald-uk** [comment]
Fixed in 7.29.0 - thanks @yogevbd
